toolchain/check/impl.cpp

```diff
@@ -33,6 +34,7 @@ static auto NoteAssociatedFunction(Context& context,
 static auto CheckAssociatedFunctionImplementation(
     Context& context, SemIR::FunctionId interface_function_id,
     SemIR::InstId impl_decl_id, Substitutions substitutions) -> SemIR::InstId {
+  CARBON_CHECK(!context.insts().Is<SemIR::FunctionType>(impl_decl_id));
```

Comment:
This seems like an unnecessary CHECK given that we have a diagnostic for the general case where `impl_decl_id` isn't a `FunctionDecl` a couple of lines below. Is it important to CHECK-fail on this, or is this a debugging aid left behind from development?
```suggestion
```

Reply:
Done (I think I added this while debugging)